scripts/build-non-release.js

```diff
@@ -0,0 +1,34 @@
+var path = require('path')
+var fs = require('fs')
+var spawn = require('child_process').spawn
+
+// If installed as an unreleased git dependency, installs the dev dependencies
+// and compiles the source. Uses yarn by default. Set `CUCUMBER_UNRELEASED_BUILD_TOOL=npm`
+// if you prefer to use npm (it's slower).
+
+const buildTool = process.env.CUCUMBER_UNRELEASED_BUILD_TOOL || 'yarn'
```

Comment:
NPM will always be present with node.js, yarn isn't guaranteed to be present.  Perhaps, test if yarn is installed before using it?

Reply:
Here's the timings with latest NPM:
```
kris git[master?] ~/src/oss/cucumber-js  
→ npm --versions
{ cucumber: '3.0.1',
  npm: '5.3.0',
  ares: '1.10.1-DEV',
  cldr: '31.0.1',
  http_parser: '2.7.0',
  icu: '59.1',
  modules: '57',
  nghttp2: '1.22.0',
  node: '8.4.0',
  openssl: '1.0.2l',
  tz: '2017b',
  unicode: '9.0',
  uv: '1.13.1',
  v8: '6.0.286.52',
  zlib: '1.2.11' }
```
and installation:
```
kris git[master] ~/src/oss/cucumber-js  
→ npm I
...
npm notice created a lockfile as package-lock.json. You should commit this file.
added 912 packages in 24.784s
```

Reply:
without yarn installed globally:
```
> cucumber-expressions@4.0.3 postinstall /Users/kris/src/oss/cucumber-js/node_modules/cucumber-expressions
> node scripts/postinstall.js

npm WARN prepublish-on-install As of npm@5, `prepublish` scripts are deprecated.
npm WARN prepublish-on-install Use `prepare` for build steps and `prepublishOnly` for upload-only.
npm WARN prepublish-on-install See the deprecation note in `npm help scripts` for more information.

> cucumber@3.0.1 prepublish /Users/kris/src/oss/cucumber-js
> yarn run build-local

sh: yarn: command not found
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn
npm ERR! cucumber@3.0.1 prepublish: `yarn run build-local`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the cucumber@3.0.1 prepublish script.
```

Reply:
Also, using shell scripts during an npm install will make the whole project not work on Windows.  Not sure if that is important to you guys.

Reply:
https://github.com/charlike/detect-installed seems like a reasonable module to use to detect the presence of yarn:

```js
const detectInstalled = require('detect-installed')
const exists = await detectInstalled('yarn')
```